lib/common/common/src/counter/hardware_accumulator.rs

```diff
@@ -1,23 +1,42 @@
+#[cfg(any(debug_assertions, test))]
+use std::sync::atomic::AtomicBool;
 use std::sync::atomic::{AtomicUsize, Ordering};
 use std::sync::Arc;
 
+use serde::Serialize;
+
 use super::hardware_counter::HardwareCounterCell;
 
 /// A "slow" but thread-safe accumulator for measurement results of `HardwareCounterCell` values.
 /// This type is completely reference counted and clones of this type will read/write the same values as their origin structure.
-#[derive(Clone)]
+#[derive(Clone, Debug, Serialize)]
 pub struct HwMeasurementAcc {
     cpu_counter: Arc<AtomicUsize>,
+
+    #[cfg(any(debug_assertions, test))]
+    applied: Arc<AtomicBool>,
 }
 
 impl HwMeasurementAcc {
     pub fn new() -> Self {
         Self::default()
     }
 
+    pub fn new_unchecked() -> Self {
```

Comment:
I see that we use this one a lot in this PR. Why can't we use the `new` variant instead for any of them?

Reply:
In the Drop-function of this type we have a check that helps us not to accidentally lose measurements in debug mode and tests.
This function makes it clear that when creating an instance of `HwMeasurementAcc`, we don't want it to have this check, which makes writing tests or hardware unrelated code easier to work with.